server/src/main/java/io/druid/server/coordinator/DruidCoordinator.java

```diff
@@ -636,10 +642,12 @@ private void stopBeingLeader()
         serviceAnnouncer.unannounce(self);
         metadataRuleManager.stop();
         metadataSegmentManager.stop();
-        leader = false;
+        lookupCoordinatorManager.stop();
       }
       catch (Exception e) {
         log.makeAlert(e, "Unable to stopBeingLeader").emit();
+      } finally {
+        leader = false;
```

Comment:
this change is out of scope here. It is a significant change in behavior if an error is encountered during the stop calls.